File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k = 0
        map = {}

        for n in nums:
            if n == val:
                continue
            map[n] = map.get(n, 0) + 1
            k += 1

        index = 0
        for key, value in map.items():
            for i in range(value):
                nums[index] = key
                index += 1

        return k

File: Data Structures & Algorithms/sort-colors/submission-1.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int check[3] = {0, 0, 0};
        for (int i: nums) ++check[i];
        int j = 0;
        for (int i = 0; i < nums.size(); ++i) {
            while (check[j] == 0) ++j;
            nums[i] = j;
            --check[j];
        }
    }
};